bootstrap/bootstrap.go

```diff
@@ -855,21 +855,30 @@ func (b *Bootstrap) executePluginHook(ctx context.Context, name string, checkout
 			return err
 		}
 
-		env, _ := p.ConfigurationToEnvironment()
-		err = b.executeHook(ctx, HookConfig{
+		envMap, err := p.ConfigurationToEnvironment()
+		if dnerr := (&plugin.DeprecatedNameErrors{}); err != nil && errors.As(err, &dnerr) {
+			b.shell.Logger.Headerf("Deprecated environment variables for plugin %s", p.Plugin.Name())
+			b.shell.Logger.Warningf(
+				"The configuration would export environment variables with the following names in a deprecated manner",
+			)
```

Comment:
it might be worth adding a "but don't worry, everything is fine for now" thing, like:
```suggestion
			b.shell.Logger.Warningf(
				"The configuration would export environment variables with the following names in a deprecated manner. We'll export both the deprecated and the replacement names, but this behaviour may change in future",
			)
```

Reply:
also: super nit:
```suggestion
			b.shell.Logger.Warningf("The configuration would export environment variables with the following names in a deprecated manner")
```